node-src/lib/getConfiguration.ts

```diff
@@ -0,0 +1,64 @@
+import { readFile } from 'jsonfile';
+import { ZodError, z } from 'zod';
+import { missingConfigurationFile } from '../ui/messages/errors/missingConfigurationFile';
+import { unparseableConfigurationFile } from '../ui/messages/errors/unparseableConfigurationFile';
+import { invalidConfigurationFile } from '../ui/messages/errors/invalidConfigurationFile';
+
+const configurationSchema = z
+  .object({
+    projectId: z.string(),
+    projectToken: z.string(),
+
+    onlyChanged: z.union([z.string(), z.boolean()]),
+    onlyStoryFiles: z.array(z.string()),
+    onlyStoryNames: z.array(z.string()),
+    untraced: z.array(z.string()),
+    externals: z.array(z.string()),
+    ci: z.boolean(),
+    debug: z.boolean(),
+    junitReport: z.union([z.string(), z.boolean()]),
+    autoAcceptChanges: z.union([z.string(), z.boolean()]),
+    exitZeroOnChanged: z.union([z.string(), z.boolean()]),
+    exitOnceUploaded: z.union([z.string(), z.boolean()]),
+    ignoreLastBuildOneBranch: z.string(),
+
+    buildScriptName: z.string(),
+    outputDir: z.string(),
+
+    storybookBuildDir: z.string(),
+    storybookBaseDir: z.string(),
+    storybookConfigDir: z.string(),
+
+    ownerName: z.string(),
+    repositorySlug: z.string(),
+  })
+  .partial();
+
+export type Configuration = z.infer<typeof configurationSchema>;
+
+export async function getConfiguration(configFile?: string) {
+  const usedConfigFile = configFile || '.chromatic.config.json';
```

Comment:
```suggestion
  const usedConfigFile = configFile || 'chromatic.config.json';
```
Very much a personal preference, but I don't like dotfiles that have an extension. It's weird. I see no reason to prefix it with a dot.

Reply:
Agreed with your reasoning @ghengeveld .